spec/functional/resource/env_spec.rb

```diff
@@ -161,6 +218,53 @@
         expect(ENV[chef_env_test_lower_case]).to eq(nil)
       end
 
+      it "should delete a user environment variable" do
+        test_resource.key_name(chef_env_test_lower_case)
+        test_resource.value(env_value1)
+        test_resource.user(env_user)
+        test_resource.run_action(:create)
+        expect(ENV[chef_env_test_lower_case]).to eq(env_value1)
+        test_resource.run_action(:delete)
+        env_obj
+        expect(@env_obj).to eq(nil)
+      end
+
+      it "should not delete an user environment variable if user are not passed" do
+        test_resource.key_name(chef_env_test_lower_case)
+        test_resource.value(env_value1)
+        test_resource.user(env_user)
+        test_resource.run_action(:create)
+        expect(ENV[chef_env_test_lower_case]).to eq(env_value1)
+        test_resource.user(default_env_user)
+        test_resource.run_action(:delete)
+        test_resource.user(env_user)
+        env_obj
+        expect(@env_obj).not_to be_nil
```

Comment:
Please use `before` block or `let` for creating the existing environment variable.